tests/api_fastapi/core_api/routes/public/test_job.py

```diff
@@ -0,0 +1,166 @@
+            scheduler_job.state = JobState.FAILED
+            session.add(scheduler_job)
+            self.scheduler_jobs.append(scheduler_job)
+            self.job_runners.append(job_runner)
+        session.commit()
+
+    def _setup_should_raise_exception_for_multiple_scheduler_on_one_host(self, session=None):
+        for _ in range(3):
+            scheduler_job = Job()
+            job_runner = SchedulerJobRunner(job=scheduler_job)
+            job_runner.job = scheduler_job
+            scheduler_job.state = State.RUNNING
+            scheduler_job.hostname = "HOSTNAME"
+            session.add(scheduler_job)
+            self.scheduler_jobs.append(scheduler_job)
+            self.job_runners.append(job_runner)
+        session.commit()
+        scheduler_job.heartbeat(heartbeat_callback=job_runner.heartbeat_callback)
+
+    @provide_session
+    def setup(self, testcase: TESTCASE_TYPE, session=None) -> None:
+        """
+        Setup testcase at runtime based on the `testcase` provided by `pytest.mark.parametrize`.
+        """
+        clear_db_jobs()
+        self.scheduler_jobs = []
+        self.job_runners = []
+        setup_method = getattr(self, f"_setup_{testcase}")
+        setup_method(session)
+
+    def teardown_method(self) -> None:
+        if self.job_runners:
+            for job_runner in self.job_runners:
+                if job_runner.processor_agent:
+                    job_runner.processor_agent.end()
+        clear_db_jobs()
+
+
+class TestGetJobs(TestJobEndpoint):
+    @pytest.mark.parametrize(
+        "testcase, query_params, expected_status_code, expected_total_entries",
+        [
+            # original testcases refactor from tests/cli/commands/test_jobs_command.py
+            (TESTCASE_ONE_SCHEDULER, {}, 200, 1),
+            (TESTCASE_ONE_SCHEDULER_WITH_HOSTNAME, {"hostname": "HOSTNAME"}, 200, 1),
+            (TESTCASE_HA_SCHEDULERS, {"limit": 100}, 200, 3),
+            (TESTCASE_IGNORE_NOT_RUNNING, {}, 200, 0),
+            (TESTCASE_MULTIPLE_SCHEDULERS_ON_ONE_HOST, {"limit": 100}, 200, 3),
+        ],
+    )
+    def test_get_jobs(
+        self, test_client, testcase, query_params, expected_status_code, expected_total_entries
+    ):
+        # setup testcase at runtime based on the `testcase` parameter
+        self.setup(testcase)
+        response = test_client.get("/public/jobs/", params=query_params)
+        assert response.status_code == expected_status_code
+        if expected_status_code != 200:
+            return
+        response_json = response.json()
+        assert response_json["total_entries"] == expected_total_entries
+
+        for idx, resp_job in enumerate(response_json["jobs"]):
+            expected_job = {
+                "id": self.scheduler_jobs[idx].id,
+                "dag_id": None,
+                "state": "running",
+                "job_type": "SchedulerJob",
+                "start_date": self.scheduler_jobs[idx].start_date.isoformat().replace("+00:00", "Z"),
```

Comment:
Nit: maybe we can use a string formatter to get the `Zulu's` string format, instead of a hard replace like this which does not seem really resilient.

Reply:
I don't think this stringformat exist yet. You need to create it and put it in the a test shared file.